aptos-move/aptos-e2e-comparison-testing/src/data_collection.rs

```diff
@@ -0,0 +1,262 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    dump_and_compile_from_package_metadata, is_aptos_package, CompilationCache, DataManager,
+    IndexWriter, PackageInfo, TxnIndex,
+};
+use anyhow::{format_err, Result};
+use aptos_framework::natives::code::PackageMetadata;
+use aptos_rest_client::Client;
+use aptos_state_view::TStateView;
+use aptos_types::{
+    state_store::{state_key::StateKey, state_value::StateValue},
+    transaction::{
+        signature_verified_transaction::SignatureVerifiedTransaction, Transaction,
+        TransactionOutput, Version,
+    },
+    write_set::TOTAL_SUPPLY_STATE_KEY,
+};
+use aptos_validator_interface::{
+    AptosValidatorInterface, DebuggerStateView, FilterCondition, RestDebuggerInterface,
+};
+use aptos_vm::{AptosVM, VMExecutor};
+use move_core_types::account_address::AccountAddress;
+use std::{
+    collections::HashMap,
+    ops::Deref,
+    path::PathBuf,
+    sync::{Arc, Mutex},
+};
+
+pub struct DataCollection {
+    debugger: Arc<dyn AptosValidatorInterface + Send>,
+    current_dir: PathBuf,
+    batch_size: u64,
+    dump_write_set: bool,
+    filter_condition: FilterCondition,
+}
+
+impl DataCollection {
+    pub fn new(
+        debugger: Arc<dyn AptosValidatorInterface + Send>,
```

Comment:
Same. Was wondering if it would make more sense to take a Stream as an input. Maybe @msmouse can provide some suggestion here.